azure/azure-how-to/instances/launch-ubuntu-images.rst

```diff
@@ -0,0 +1,130 @@
+Launch Ubuntu Images on Azure
+=============================
+
+This documentation is based on the `official Azure documentation <https://learn.microsoft.com/en-us/azure/virtual-machines/linux/quick-create-cli>`_
+for creating Linux virtual machines with the Azure CLI.
+
+Prerequisites
+-------------
+
+- A Microsoft Azure account
+- `Azure Command-Line Interface <https://learn.microsoft.com/en-us/cli/azure/>`_
+- The URN for a Ubuntu image on Azure to launch (see :ref:`Find Ubuntu images on Azure`)
+
+Launch a Ubuntu Image
+---------------------
+
+To launch a Ubuntu Image, you will need to create a resource group and create a virtual machine using the Ubuntu image
+you have selected.
+
+Create the Resource Group
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Define variables to select the resource group name and location to deploy:
+
+.. code::
+
+    RESOURCE_GROUP_NAME="ubuntu-vm-rg"
+    LOCATION="eastus"
+
+Create a resource group using the `az group create <https://learn.microsoft.com/en-us/cli/azure/group?view=azure-cli-latest#az-group-create>`_ command:
+
+.. code::
+
+    az group create \
+        -n $RESOURCE_GROUP_NAME \
+        -l $LOCATION
+
+Create the Virtual Machine
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Create a virtual machine using the `az vm create <https://learn.microsoft.com/en-us/cli/azure/vm?view=azure-cli-latest#az-vm-create>`_ command.
+
+Depending on the type of Ubuntu image you have selected, additional arguments may be required to correctly launch
+the image. Most Ubuntu images can be launched with the default arguments:
```

Comment:
Can we create a separate section here mentioning all the images that can be launched with the default arguments? So under "Create the Virtual Machine" we'll have:
- Default configuration (say)
- Security type - Trusted Launch
- Security type - Confidential VM
- And any other specific configurations (FIPS compliant images? CIS hardened images?)

Reply:
It is safe to say every single Ubuntu image (minus Ubuntu Anbox, which is in its own form of development hell because of Azure) can be launched with the default arguments, it's just that the image will not behave as one would expect. This is where I worry the documentation will become overly muddied by the matrix of possible configurations, e.g.:

Image A can be booted using defaults and Trusted Launch, except the Gen1X64 sku which does not support Trusted Launch, etc.

In my mind the focus for a how-to launch Ubuntu on Azure is a starting point, and not a detailed matrix of all combinations where the reader will be confused by the overhead of extraneous items. Covering all of the Ubuntu images which are compatible with individual configurations will also be high maintenance.

What I would prefer is a follow up documentation explaining the difference in capabilities between CoreVM and AzureVirtualMachines (abstracting these technical differences and instead focus on that the images we publish as AzureVirtualMachines are not always feature parity with CoreVM - e.g. TrustedLaunch, although this is currently being resolved).

If we do want exhaustive coverage of what images support what Azure features, we should instead construct a table/matrix of offer:plan against Azure features and what is supported. Maintaining it as a pure text document will be difficult to navigate and maintain.

Reply:
Sounds good to me. Could you please create a Jira ticket for this follow-up documentation (CoreVM vs AzureVirtualMachines)? And we can take a call on whether the exhaustive coverage is needed or not.